code/Test_definitions/device-reachability-status-subscriptions.feature

```diff
@@ -0,0 +1,225 @@
+    Then the response code is 201 
+    When the request "deleteReachabilityStatusSubscription" is sent
+    Then the response code is 202 or 204	
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
+    And the response property "$.terminationReason" is "SUBSCRIPTION_DELETED"
+
+
+############### Error response scenarios ###########################
+
+  @reachability_status_subscriptions_13_Create_reachability_status_subscription_with_invalid_parameter
+  Scenario:  Create subscription with invalid parameter
+    Given the request body is not compliant with the schema "/components/schemas/SubscriptionRequest"
+    When the  request "createSubscription" is sent 
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+@reachability_status_subscriptions_14_creation_of_subscription_with_expiry_time_in_past
+  Scenario: Expiry time in past
+    Given a valid subscription request body 
+    And request body property "$.subscriptionexpiretime" in past
+    When the  request "createSubscription" is sent 
+    Then the response code is 400
+    And the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_ARGUMENT"
+    And the response property "$.message" contains a user friendly text
+
+ @reachability_status_subscription_15_invalid_protocol
+   Scenario: subscription creation with invalid protocol
+    Given a valid subscription request body 
+    And  the request property "$.protocol" is not set to "HTTP"
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_PROTOCOL"
+    And the response property "$.message" contains a user friendly text
+
+@reachability_status_subscription_16_invalid_credential_type
+   Scenario: subscription creation with invalid credential type
+    Given a valid subscription request body 
+    And the request property "$.credentialType" is not "ACCESSTOKEN"
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_CREDENTIAL"
+    And the response property "$.message" contains a user friendly text
+	
+@reachability_status_subscription_17_invalid_access_token_type
+   Scenario: subscription creation with invalid access token type 
+    Given a valid subscription request body 
+    And the request property "$.accessTokenType" is not "bearer"
+    When the request "createSubscription" is sent
+    Then the response property "$.status" is 400
+    And the response property "$.code" is "INVALID_TOKEN"
```

Comment:
For alignement with geofencing:
`And the response property "$.code" is "INVALID_TOKEN" or "INVALID_ARGUMENT"`